examples/seq2seq/distillation.py

```diff
@@ -126,16 +149,21 @@ def _step(self, batch):
         # assert is_frozen(self.teacher) copied_decoder_layers
         pad_token_id = self.tokenizer.pad_token_id
         input_ids, src_mask, labels = batch["input_ids"], batch["attention_mask"], batch["labels"]
+        if isinstance(self.teacher, T5ForConditionalGeneration):
+            teacher_decoder_input_ids = self.teacher._shift_right(labels)
+        else:
+            teacher_decoder_input_ids = shift_tokens_right(labels, pad_token_id)
+
         if isinstance(self.model, T5ForConditionalGeneration):
-            decoder_input_ids = self.model._shift_right(labels)
+            student_decoder_input_ids = self.model._shift_right(labels)
```

Comment:
if input_ids are the same, is the `student_decoder_input_ids`/`teacher_decoder_input_ids` distinction relevant?

Reply:
yes, I wondered about it too. It probably doesn't make a difference. Problem is the API makes it look as if it's (also) dependent on the model since `_shift_right` is being called on the model. Should I just get rid of this and have just one set of `input_ids`? I'll test to see that that works.

Reply:
yes!

Reply:
done